server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -53,10 +52,16 @@ const updateFilterConditionsForReadyToPay = async (where, include, host): Promis
   }
 
   if (checkTaxForms) {
-    const expensesSubjectToTaxForm = expenses.filter(e => !TAX_FORM_IGNORED_EXPENSE_TYPES.includes(e.type));
+    const expensesSubjectToTaxForm = expenses.filter(expenseMightBeSubjectToTaxForm);
     if (expensesSubjectToTaxForm.length > 0) {
-      const expensesIdsSubjectToTaxForm = expensesSubjectToTaxForm.map(expense => expense.id);
-      expensesIdsPendingTaxForms = await queries.getTaxFormsRequiredForExpenses(expensesIdsSubjectToTaxForm);
+      const expenseIds = expensesSubjectToTaxForm.map(expense => expense.id);
+      const requiredLegalDocs = await loaders.Expense.taxFormRequiredBeforePayment.loadMany(expenseIds);
+      for (let i = 0; i < requiredLegalDocs.length; i++) {
+        if (requiredLegalDocs[i]) {
+          expensesIdsPendingTaxForms.add(expenseIds[i]);
+        }
+      }
```

Comment:
Suggestion for readability:
```suggestion
      requiredLegalDocs.forEach((required, i) => {
        if (required) {
          expensesIdsPendingTaxForms.add(expenseIds[i]);
        }
      });
```